source/common/upstream/upstream_impl.cc

```diff
@@ -253,6 +256,30 @@ ClusterLoadReportStats ClusterInfoImpl::generateLoadReportStats(Stats::Scope& sc
   return {ALL_CLUSTER_LOAD_REPORT_STATS(POOL_COUNTER(scope))};
 }
 
+// TODO(alanconway): dummy factory context, how do we implement this?  Some of
```

Comment:
I think the only fields that are listener-specific are listenerScope(), listenerMetadata(), and possibly drainDecision(). 

listenerScope() is only used in a few places in the code-base. We may be able to leave it NOT_IMPLEMENTED here, or maybe rename it to ownerScope() or something more generic, and pass it the cluster's scope. 

listenerMetadata() isn't called anywhere except one test. Can it be deleted? Or am I missing something?